TACC-Cloud/hazmapper
# task/WG-219: Adding GET hook for grabbing the username

## Overview: ##
Add react-query GET in our react/src/hooks to get user name

## PR Status: ##


## Related Jira tickets: ##

* [WG-219](https://tacc-main.atlassian.net/browse/WG-219)

## Summary of Changes: ##

- Added a useAuthenticatedUser hook to retrieve username from Tapis
- Previously this was being done in redux
- Added a custom transform option to the useGet params in request.ts. This will help us transform API responses into the desired structure in the future if we have another case where the response data (name) needs to be changed to something more descriptive for us (username). 
- Added some temp UI to showcase the useAuthenticatedUser hook in action. This can be seen on the Main Menu, welcoming the authenticated user. We will get rid of this once the Nav-bar and Project Listing PRs are merged into master. 
- Between this PR (WG-219) and WG-74, these two PRs will need to be combined in the near future. WG-74's MapModal needs to take advantage of this useAuthenticatedUsesr hook. Which also includes updating MapModal's unit test to get rid of redux and use this useName hook instead. 

## Testing Steps: ##
1. Run npm ci and npm run dev
2. Go to localhost:4200
3. On the Main Menu you should see a "Welcome" message with your username, upon login
4. Compare with screenshot below

## UI Photos:
![Screenshot 2024-01-22 at 2 56 04 PM](https://github.com/TACC-Cloud/hazmapper/assets/50084480/64cf43f3-9650-4eed-a92c-9bf51031ad56)

## Notes: ##
